File: packages/elixir-sdk/test/adapters/runtime/observability_test.exs
defmodule TreeDxSdk.ObservabilityAdapterTest do
  use ExUnit.Case, async: true

  test "all constructs expected request" do
    {:ok, pid} = TreeDxSdk.Test.MockTransport.start_link()
    client = TreeDxSdk.Test.MockTransport.client(pid)
    TreeDxSdk.Observability.health(client)
    TreeDxSdk.Observability.ready(client)
    TreeDxSdk.Observability.deep_health(client)
    TreeDxSdk.Observability.metrics(client)

    assert Enum.any?(
             TreeDxSdk.Test.MockTransport.requests(pid),
             &(&1.method == :get and &1.path == "/api/v1/metrics")
           )
  end
end

File: packages/ts-sdk/test/adapters/runtime/observability.test.ts
import { describe, expect, it } from 'vitest';
import { ObservabilityAdapter } from '../../../src/treedx/index.js';
import { MockTransport } from '../mock.js';

describe('ObservabilityAdapter', () => {
  it('constructs observability endpoints', async () => {
    const transport = new MockTransport();
    const adapter = new ObservabilityAdapter({ transport });
    await adapter.health();
    await adapter.ready();
    await adapter.deepHealth();
    await adapter.metrics();
    expect(transport.requests.map((r) => `${r.method} ${r.path}`)).toEqual([
      'GET /api/v1/health',
      'GET /api/v1/ready',
      'GET /api/v1/health/deep',
      'GET /api/v1/metrics'
    ]);
  });
});
